tests/test_translations.h

```diff
@@ -0,0 +1,75 @@
+/*************************************************************************/
+/*  test_translations.h                                                  */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_TRANSLATIONS_H
+#define TEST_TRANSLATIONS_H
+
+#include "core/io/resource_loader.h"
+#include "core/os/memory.h"
+#include "core/string/translation.h"
+#include "editor/import/resource_importer_csv_translation.h"
+
+namespace TestTranslations {
+
+TEST_CASE("[Translation] CSV import") {
+	TranslationServer *translation_server = nullptr;
+	translation_server = memnew(TranslationServer);
+	translation_server->setup(); //register translations, load them, etc.
+	translation_server->load_translations();
```

Comment:
I'm a bit concerned about instantiation of another `TranslationServer` because it's likely already setup in `test_main()`. Instantiating `TranslationServer` may lead to an override of the main singleton.

You can try to remove this, include translation server and use `TranslationServer::get_singleton()` instead, and it should work I think.

Reply:
I've tried to remove it but it looks like singleton wasn't initialized.